Formula/grafana.rb

```diff
@@ -0,0 +1,154 @@
+      bin.install "bin/grafana-server"
+      (bin/"grafana").write(env_script)
+      chmod 0755, bin/"grafana"
+      (etc/"grafana").mkpath
+      cp("conf/sample.ini", "conf/grafana.ini.example")
+      unless (etc/"grafana/grafana.ini").exist?
+        etc.install "conf/sample.ini" => "grafana/grafana.ini"
+      end
+      etc.install "conf/grafana.ini.example" => "grafana/grafana.ini.example"
+      pkgshare.install Dir["conf", "vendor"]
+      pkgshare.install "public_gen" => "public"
+    end
+  end
+
+  def post_install
+    (var/"log/grafana").mkpath
+    (var/"lib/grafana/plugins").mkpath
+  end
+
+  def env_script
+    <<-EOS.undent
+      #!/usr/bin/env bash
+      DAEMON=grafana-server
+      EXECUTABLE=#{bin/"grafana-server"}
+      CONFIG=#{etc}/grafana/grafana.ini
+      HOMEPATH=#{pkgshare}
+      LOGPATH=#{var}/log/grafana
+      DATAPATH=#{var}/lib/grafana
+      PLUGINPATH=#{var}/lib/grafana/plugins
+
+      case "$1" in
+      start)
+        $EXECUTABLE --config=$CONFIG --homepath=$HOMEPATH cfg:default.paths.logs=$LOGPATH cfg:default.paths.data=$DATAPATH cfg:default.paths.plugins=$PLUGINPATH 2> /dev/null &
+        [ $? -eq 0 ] && echo "$DAEMON started"
+      ;;
+      stop)
+        killall $DAEMON
+        [ $? -eq 0 ] && echo "$DAEMON stopped"
+      ;;
+      restart)
+        $0 stop
+        $0 start
+      ;;
+      *)
+        echo "Usage: $0 (start|stop|restart)"
+      ;;
+      esac
+    EOS
+  end
+
+  plist_options :manual => "grafana start"
+
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>KeepAlive</key>
+        <dict>
+          <key>SuccessfulExit</key>
+          <false/>
+        </dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>ProgramArguments</key>
+        <array>
+          <string>#{opt_bin}/grafana-server</string>
+          <string>--config</string>
+          <string>#{etc}/grafana/grafana.ini</string>
+          <string>--homepath</string>
+          <string>#{opt_pkgshare}</string>
+          <string>cfg:default.paths.logs=#{var}/log/grafana</string>
+          <string>cfg:default.paths.data=#{var}/lib/grafana</string>
+          <string>cfg:default.paths.plugins=#{var}/lib/grafana/plugins</string>
+        </array>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>WorkingDirectory</key>
+        <string>#{var}/share/grafana</string>
+        <key>StandardErrorPath</key>
+        <string>#{var}/log/grafana/grafana-stderr.log</string>
+        <key>StandardOutPath</key>
+        <string>#{var}/log/grafana/grafana-stdout.log</string>
+        <key>SoftResourceLimits</key>
+        <dict>
+          <key>NumberOfFiles</key>
+          <integer>10240</integer>
+        </dict>
+      </dict>
+    </plist>
+   EOS
+  end
+
+  test do
+    require "pty"
```

Comment:
Is it particularly worthwhile to do so here? PTY is in the ruby stdlib, so it's not a gem dependency that's unlikely to be installed. I had considered `fork` when I was writing this test, but using `spawn` seemed a lot more straightforward (but see below) as far as being able to run grafana long enough to see if it successfully started, kill the process, and read its output. With `fork` it still needs to get the standard out from the child and kill the running process (because grafana won't stop on its own). I can do it that way, but I'm not sure it's actually better.

Originally here I used `Process.spawn`, which Worked On My Machine™ running Mavericks, but failed when running on the homebrew jenkins server when it was building it for Mavericks (but worked fine on Yosemite and El Capitan). It turned out that my system ruby on this Mavericks box had been upgraded to Ruby 2.0, while the brew jenkins Maverics was using 1.8.7 as its system ruby. Unfortunately, `Process.spawn` isn't available under 1.8.7.

I had changed it to use `Process.spawn` for Yosemite and El Capitan and `PTY.spawn` for Mavericks, but decided that it was less confusing to just use `PTY.spawn` for all three of them.

That said, once El Capitan's successor comes out & homebrew drops Mavericks then `Process.spawn` would work just fine.